C7GameData/Tile.cs

```diff
@@ -261,6 +443,32 @@ public static (int, int) toCoordDiff(this TileDirection dir)
 			}
 		}
 
+		// Essentially finds the two other tiles that neighbor BOTH diagonal tiles
+		public static (TileDirection, TileDirection) diagonalComplementary(this TileDirection dir)
+		{
+			switch(dir)
+			{
+				case TileDirection.NORTH: return (TileDirection.NORTHEAST, TileDirection.NORTHWEST);
+				case TileDirection.EAST: return (TileDirection.SOUTHEAST, TileDirection.NORTHWEST);
+				case TileDirection.SOUTH: return (TileDirection.SOUTHEAST, TileDirection.SOUTHWEST);
+				case TileDirection.WEST: return (TileDirection.NORTHWEST, TileDirection.SOUTHWEST);
+				default: throw new ArgumentOutOfRangeException("TileDirection is not Diagonal");
```

Comment:
I have mixed feelings on this solution to giving an invalid direction.  Crashing the game is bad.  We had a bug in early 2022 where a method would throw an uncaught exception if it got bad data, and it did sometimes, and the game would mysteriously crash.  This doesn't look as likely to cause that issue but all it would take is someone passing NORTHEAST into this method because they are thinking of Civ4 tiles where NORTHEAST is diagonal.

It may, or may not, be better to instead log an error, but to return a value (a tuple of the same direction passed in, in both parts of the tuple?) rather than crash.  Is the resulting bug in game logic worse than a crash?  It varies, but IMO crashes are the quickest way to negative reviews and players quitting a game (or just never restarting it after the crash).